processing/src/main/java/io/druid/query/lookup/RegisteredLookupExtractionFn.java

```diff
@@ -49,93 +52,109 @@ public static RegisteredLookupExtractionFn create(
   )
   {
     Preconditions.checkArgument(lookup != null, "`lookup` required");
-    final LookupExtractorFactory factory = manager.get(lookup);
-    Preconditions.checkNotNull(factory, "lookup [%s] not found", lookup);
-    return new RegisteredLookupExtractionFn(
-        new LookupExtractionFn(
-            factory.get(),
-            retainMissingValue,
-            replaceMissingValueWith,
-            injective,
-            optimize
-        ),
-        lookup
-    );
+    this.manager = manager;
+    this.replaceMissingValueWith = replaceMissingValueWith;
+    this.retainMissingValue = retainMissingValue;
+    this.injective = injective;
+    this.optimize = optimize;
+    this.lookup = lookup;
   }
 
   @JsonProperty("lookup")
   public String getLookup()
   {
-    return name;
+    return lookup;
   }
 
   @JsonProperty("retainMissingValue")
   public boolean isRetainMissingValue()
   {
-    return delegate.isRetainMissingValue();
+    return retainMissingValue;
   }
 
   @JsonProperty("replaceMissingValueWith")
   public String getReplaceMissingValueWith()
   {
-    return delegate.getReplaceMissingValueWith();
+    return replaceMissingValueWith;
   }
 
   @JsonProperty("injective")
   public boolean isInjective()
   {
-    return delegate.isInjective();
+    return injective;
   }
 
   @JsonProperty("optimize")
   public boolean isOptimize()
   {
-    return delegate.isOptimize();
+    return optimize;
   }
 
   @Override
   public byte[] getCacheKey()
   {
-    return delegate.getCacheKey();
+    try (final ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
```

Comment:
I know it makes them, but with BAOS I don't have to MANAGE them. I'll change the code and you can tell me if it looks easier to manage